src/Microsoft.ML.CpuMath/CpuMathUtils.netstandard.cs

```diff
@@ -20,72 +21,52 @@ public static int GetVectorAlignment()
 
-        public static void AddScale(float a, float[] src, float[] dst, int dstOffset, int count) => SseUtils.AddScale(a, src, dst, dstOffset, count);
+        public static void AddScaleCopy(float a, ReadOnlySpan<float> src, ReadOnlySpan<float> dst, Span<float> res, int count) => SseUtils.AddScaleCopy(a, src, dst, res, count);
 
-        public static void AddScale(float a, float[] src, int[] indices, float[] dst, int count) => SseUtils.AddScale(a, src, indices, dst, count);
+        public static void Add(ReadOnlySpan<float> src, Span<float> dst, int count) => SseUtils.Add(src, dst, count);
 
-        public static void AddScale(float a, float[] src, int[] indices, float[] dst, int dstOffset, int count) => SseUtils.AddScale(a, src, indices, dst, dstOffset, count);
+        public static void Add(ReadOnlySpan<float> src, ReadOnlySpan<int> indices, Span<float> dst, int count) => SseUtils.Add(src, indices, dst, count);
 
-        public static void AddScaleCopy(float a, float[] src, float[] dst, float[] res, int count) => SseUtils.AddScaleCopy(a, src, dst, res, count);
+        public static void MulElementWise(ReadOnlySpan<float> src1, ReadOnlySpan<float> src2, Span<float> dst, int count) => SseUtils.MulElementWise(src1, src2, dst, count);
 
-        public static void Add(float[] src, float[] dst, int count) => SseUtils.Add(src, dst, count);
+        public static float Sum(ReadOnlySpan<float> src) => SseUtils.Sum(src);
 
-        public static void Add(float[] src, int[] indices, float[] dst, int count) => SseUtils.Add(src, indices, dst, count);
+        public static float SumSq(ReadOnlySpan<float> src) => SseUtils.SumSq(src);
 
-        public static void Add(float[] src, int[] indices, float[] dst, int dstOffset, int count) => SseUtils.Add(src, indices, dst, dstOffset, count);
+        public static float SumSq(float mean, ReadOnlySpan<float> src) => SseUtils.SumSq(mean, src);
 
-        public static void MulElementWise(float[] src1, float[] src2, float[] dst, int count) => SseUtils.MulElementWise(src1, src2, dst, count);
+        public static float SumAbs(ReadOnlySpan<float> src) => SseUtils.SumAbs(src);
 
-        public static float Sum(float[] src, int count) => SseUtils.Sum(src, count);
+        public static float SumAbs(float mean, ReadOnlySpan<float> src) => SseUtils.SumAbs(mean, src);
 
-        public static float Sum(float[] src, int offset, int count) => SseUtils.Sum(src, offset, count);
+        public static float MaxAbs(ReadOnlySpan<float> src) => SseUtils.MaxAbs(src);
 
-        public static float SumSq(float[] src, int count) => SseUtils.SumSq(src, count);
+        public static float MaxAbsDiff(float mean, ReadOnlySpan<float> src) => SseUtils.MaxAbsDiff(mean, src);
 
-        public static float SumSq(float[] src, int offset, int count) => SseUtils.SumSq(src, offset, count);
```

Comment:
>SumSq [](start = 28, length = 5)

did these end up not necessary?

I wonder why were they added in the first place then. Maybe some TLC code will rely on them?

Reply:
They are no longer necessary because the `Span` parameter takes care of needing to pass (and verify) `offset` and `count`.  (Span is a pointer and a count). If you have an array, and need to offset into it, or limit the count, you can call `array.AsSpan(start, length)` and now you can call the one overload that takes a Span.

This helps make the CpuMathUtils class consistent. Previously, some methods had overloads that take an `offset`, some didn't. Some had multiple arrays and only one of those arrays had a corresponding `offset` parameter, etc.

> I wonder why were they added in the first place then

We never had Span before. 😉 

> Maybe some TLC code will rely on them?

If that's the case, they can call `SumSq(array.AsSpan(offset, count))` instead of `SumSq(array, offset, count)`.

Reply:
I noticed some of the AVX intrinsics were not used when I re-wrote them in pure C# when creating a OSX build about 2 yrs ago. I assumed at the time either they were for future use (completeness of the intrinsics set) or I simply wasn't calling code where they were used.

Reply:
I've noticed the same. In fact, the whole "AVX" C++ code is not being called at all in ML.NET.

Reply:
https://github.com/dotnet/machinelearning/issues/780 we have issue for that.